### ISSUE TITLE
bug(datepicker): NVDA announces entire week of dates when dialog opens

### ISSUE THREAD

**Issue by steven-curran-ub** [angular/components#27228]
### Is this a regression?

- [ ] Yes, this behavior used to work in the previous version

### Description

Posting for our team's accessibility expert:

Our accessibility expert found that in NVDA 2023.1, when a user opens the datepicker dialog, the screen reader announces the entire next week of dates from the focus date. 
**We did not observe this behavior with JAWS 2023 or VoiceOver on MacOS 13.4.**

Related, turning off the default "Automatic Say All on Page Load" setting in NVDA (under Preferences > Browse Mode) does "remedy" the issue. However, our accessibility team is wary of asking users to change a default setting in NVDA for this.
Based on that, internally we have proposed this may be an NVDA problem, but we were asked to start here since NVDA is so widely used.

### Reproduction

StackBlitz link: https://stackblitz.com/edit/ahmyck
Steps to reproduce:
1. Click into the date field
2. Tab over to the dialog-open button
3. Hit enter and listen to the entire screen reader output (or check something like Speech Viewer for NVDA)

Notice the announcement of the entire next week of dates.

### Expected Behavior

Expected less verbosity and for only the focused date/cell to be announced.

### Actual Behavior

Currently, the entire next week of dates will be announced by the screen reader after the currently focused cell.

NVDA 2023.1 output (italics indicate unexpected readout):
> Choose a date  dialog
> table
> June 2, 2023  row 3  Friday  column 6
> table  with 7 rows and 7 columns  clickable  row 3  Friday  column 6  June 2, 2023  toggle button  not pressed    current date
> Choose a date
> toggle button  not pressed    current date  June 2, 2023
> _Saturday  column 7  toggle button  not pressed    June 3, 2023_
> _row 4  Sunday  column 1  toggle button  not pressed    June 4, 2023_
> _Monday  column 2  toggle button  not pressed    June 5, 2023_
> _Tuesday  column 3  toggle button  not pressed    June 6, 2023_
> _Wednesday  column 4  toggle button  not pressed    June 7, 2023_
> _Thursday  column 5  toggle button  not pressed    June 8, 2023_
> _Friday  column 6  toggle button  not pressed    June 9, 2023_
> _Saturday  column 7  toggle button  not pressed    June 10, 2023_
> _row 5  Sunday  column 1  toggle button  not pressed    June 11, 2023_

### Environment

- Angular: 13.x, 14.x, 15.x, 16.x
- CDK/Material: 13.x, 14.x, 15.x, 16.x
- Browser(s): Chrome (114), Firefox (112) 
- Operating System (e.g. Windows, macOS, Ubuntu): Windows 10, Windows 11
- Screen readers: NVDA 2023.1

**Comment by zarend** [comment]
Hello folks,

Thank you for reporting accessibility concern. We need more information before we can accept this issue report.

> Our accessibility expert found that in NVDA 2023.1

Do you have an accessibility requirement thats not being meet or an explanation why this is a problem for users? this may be just how NVDA works.

Best Regards,

Zach

**Comment by steven-curran-ub** [comment]
Hey, I appreciate you guys following up on this.

Unfortunately, the only information I have from when this was cited as an issue was effectively just "it was a poor experience and could cause confusion". 

Given that this issue is a bit older now and we only saw this with NVDA, I just retested this using the latest version of NVDA 2023.3.3, and it's no longer reading out the full next week of date buttons.

I'm going to go ahead and close this since I believe it was indeed an NVDA quirk.

Sorry about that guys!
